fe/fe-core/src/main/java/com/starrocks/connector/jdbc/JDBCMetadata.java

```diff
@@ -68,27 +73,36 @@ public JDBCMetadata(Map<String, String> properties, String catalogName) {
             throw new StarRocksConnectorException(properties.get(JDBCResource.DRIVER_CLASS) + " not support yet");
         }
         checkAndSetSupportPartitionInformation();
+        createMetaAsyncCacheInstances(properties);
     }
 
-    public Connection getConnection() throws SQLException {
-        return DriverManager.getConnection(properties.get(JDBCResource.URI),
-                properties.get(JDBCResource.USER), properties.get(JDBCResource.PASSWORD));
+    private void createMetaAsyncCacheInstances(Map<String, String> properties) {
+        partitionNamesCache = new JDBCMetaCache<>(properties, false);
+        tableIdCache = new JDBCMetaCache<>(properties, true);
+        tableInstanceCache = new JDBCMetaCache<>(properties, false);
+        partitionInfoCache = new JDBCMetaCache<>(properties, false);
     }
 
     public void checkAndSetSupportPartitionInformation() {
         try (Connection connection = getConnection()) {
             schemaResolver.checkAndSetSupportPartitionInformation(connection);
         } catch (SQLException e) {
-            throw new StarRocksConnectorException(e.getMessage());
+            throw new StarRocksConnectorException(
+                    "check and set support partition information for JDBC catalog fail!", e);
         }
     }
 
+    public Connection getConnection() throws SQLException {
+        return DriverManager.getConnection(properties.get(JDBCResource.URI),
+                properties.get(JDBCResource.USER), properties.get(JDBCResource.PASSWORD));
+    }
+
     @Override
     public List<String> listDbNames() {
         try (Connection connection = getConnection()) {
             return Lists.newArrayList(schemaResolver.listSchemas(connection));
         } catch (SQLException e) {
-            throw new StarRocksConnectorException(e.getMessage());
+            throw new StarRocksConnectorException("refresh db names cache for JDBC catalog fail!", e);
```

Comment:
if that user dose not enable cache and throw this exception is puzzling

Reply:
done